### ISSUE TITLE
Edge Cases Thread

### ISSUE THREAD

**Issue by vhyrro** [nvim-neorg/norg-specs#24]
# All the Edge Cases

The specification isn't perfect, and as such this thread exists for any other parser writers to ask questions or note inconsistencies/ambiguities in the spec.

cc @boltlessengineer :D

**Comment by boltlessengineer** [comment]
whitespace/end-of-line rules in link destination:

```
{/ pa
th/to
/file.txt}

{:pa th/t\ o/file:}

{:path/t\
o/file:}
```

**Comment by boltlessengineer** [comment]
### link destinations should be case-insensitive

I couldn't find this in doc, but Neorg seems to be working like this.

```
$ Paragraph Break
description

... {$ Paragraph Break} ... %this is valid%
... {$ paragraph break} ... %this is also valid%
```

**Comment by boltlessengineer** [comment]
What happens when we put carryover tag inside attached modifiers? Considering the precedence, should it break the attached modifier? Or as line-ending inside attached modifier can be ignored, carryover tag inside attached modifier is just invalid?

```
*bold
+color red
text*
```

**Comment by boltlessengineer** [comment]
Does _terminating slide with a paragraph break_ means it is only terminating that list item but not a whole list?

```
- :
  paragraph

- is this same list? or another list?
```

**Comment by boltlessengineer** [comment]
Anchor link inside anchor link

```
[[my link]]
```

I think we can solve this edge case by converting to syntax sugar for wikilinks (automatically converted to `{? my link}`)
Wiki-links are used a lot in many workflows so having extra syntax sugar made up with lowercases would be really useful.

**Comment by boltlessengineer** [comment]
Free form attached modifier inside headings

```
* Contextual `|` Delimiter
  The pipe (`|`) character ...
```

Cause `|` opens free-form attached modifier, these two lines are part of single paragraph segment inside header

**Comment by vhyrro** [comment]
Extra: spec doesn't define what sort of objects are allowed after a single definition (only a paragraph? or any single complex object?)

**Comment by boltlessengineer** [comment]
escape sequence can used to escape non-punctuation character, is this intended behavior?

```
\* - this is common usecase of escape sequence

\a - and this is also escape sequence according to spec
```

**Comment by boltlessengineer** [comment]
### Injection to link location

```
* evil link injection} {https://example.com

link to {* evil link injection} {https://example.com}[that heading]
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Should spec disallow `{` and `}` characters inside link location?

**Comment by boltlessengineer** [comment]
### Structable's prefix inside attached modifiers?

```
*bold
- 
word*
```

If there is no following whitespace in line 2, this would be valid attached modifier, right?
But if line 2 is `- ` (`-` followed by whitespace,)
is this _valid attached modifier_ or _syntax error for unordered list_?

**Comment by vhyrro** [comment]
Since detached modifiers have a higher precedence than attached modifiers it becomes an unclosed `*bold`, a syntax error for the unordered list and then `word*`

**Comment by vhyrro** [comment]
> escape sequence can be used to escape non-punctuation character, is this intended behavior?
> 
> ```
> \* - this is common usecase of escape sequence
> 
> \a - and this is also escape sequence according to spec
> ```

Yep, it is! It's for the sake of consistency. It won't do anything, in fact it will just return an `a` as is :)
